tap_trello/streams.py

```diff
@@ -76,59 +76,46 @@ def _get_window_state(self):
         window_state = {}
 
         start_date = self.config['start_date']
-        window_start_bookmark = (singer.get_bookmark(self.state, self.stream_id, 'window_start') or
-                                 singer.get_bookmark(self.state, self.stream_id, 'last_sync_end') or
-                                 start_date)
-        window_start = utils.strptime_to_utc(max(window_start_bookmark, start_date))
 
-        window_sub_end = singer.get_bookmark(self.state, self.stream_id, 'window_sub_end')
-        window_sub_end = window_sub_end and utils.strptime_to_utc(window_sub_end)
+        window_start_bookmark = (singer.get_bookmark(self.state, self.stream_id, 'window_start'))
+        window_start = utils.strptime_to_utc(max(window_start_bookmark, start_date))
 
-        next_window_start = singer.get_bookmark(self.state, self.stream_id, 'next_window_start')
-        next_window_start = next_window_start and utils.strptime_to_utc(next_window_start)
+        sub_window_end = singer.get_bookmark(self.state, self.stream_id, 'sub_window_end')
+        sub_window_end = sub_window_end and utils.strptime_to_utc(sub_window_end)
 
-        if window_sub_end is not None and next_window_start is None:
-            raise Exception("Invalid State: window_sub_end provided without a next_window_start.")
+        window_end = singer.get_bookmark(self.state, self.stream_id, 'window_end') # TODO: how should this consider end_date?
```

Comment:
Seems like this comment is handled, `end_date` is the default in `on_window_started`.

Reply:
I think if `window_end < sub_window_end`, then I think it could be considered as a new window. Either by making sub_window_end null or setting them equal to the `min(window_end, sub_window_end)`